### DISCUSSION TITLE
Imbalanced Dataset Training

### DISCUSSION THREAD

**Discussion by Spawnfile** [ultralytics/yolov5#9720] [Q&A] [answered]
Hi everyone,

I've downloaded OpenImagesV6 dataset with multiple instances to train a multi-class object detector on yolov5 model. Unfortunately, due to the imbalance instance volumes on these different classes one of them did not learn anything after training was done. Because of I use a strict dataset, I can not increase instance numbers of dataset so there is 2 things in my mind to solve this problem.

---

**Instance counts**
```
class_1 = 1M
class_2 = 200k
class_3 = 1M
```
---

1) The training process I talked about was a `from scratch` training, so re-training the model with pre-trained weights with same dataset
2) Implementing a torch sampler depends on percentage of instance numbers like RandomWeightSampler

Which one should I choose or is there another ways to get over with this problem ?

Best
Alper

**Comment by glenn-jocher** [comment, marked as answer]
@Spawnfile COCO has far worse class imbalances and trains fine. Usage examples show training from pretrained weights, so start there.
